### PR TITLE
docs: explain our 100% coverage requirement

### PR DESCRIPTION
**Pull request by rowanmanning**
I've added a section to the contributing guide explaining why we expect
100% unit test coverage and how to get around it if the code you're
writing does not need to be covered.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -69,14 +70,31 @@ As with ESLint, we check types in pull requests and errors will block merging, s
...
+```js
+/* istanbul ignore next */
+function example() { console.log('this is not covered'); }
+```
+
+This is better than dropping the required coverage because:
+
+  * We have to make an active and conscious decision to drop a piece of code below 100% coverage
+
+  * The fact that code isn't covered is _documented_ in the code
```
Comment (JSRedondo) [inline docs/contributing.md:94]: Great decision. The issue about the 100% branch coverage is that sometimes it requires additional tests for default parameters.
`function example(a={}, b=true){}`

**Review by JSRedondo — APPROVED** [review]

File: docs/contributing.md [hunk 2]
```diff
@@ -69,14 +70,31 @@ As with ESLint, we check types in pull requests and errors will block merging, s
...
 
+#### Coverage
+
+We intentionally fail the unit tests if coverage drops below 100%. This library is meant to help our applications be more reliable and so it's important that we cover as many edge cases as possible. If you have a valid reason for some code not to be covered, e.g. an empty function as a default parameter, then use code comments to disable coverage for that line or block:
+
+```js
+/* istanbul ignore next */
+function example() { console.log('this is not covered'); }
+```
+
```
Comment (JSRedondo) [inline docs/contributing.md:89]: More on this, maybe add this reference to the documentation
[Ignoring code for code coverage](https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md)

**Review by JSRedondo — APPROVED** [review]

File: docs/contributing.md (hunk 2 above)
Comment (rowanmanning) [reply docs/contributing.md:89]: Good idea 👍
